src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1865,9 +1865,11 @@ private static Guid GenerateGuidFromName(string name)
                     {
                         dataType = Enum.GetUnderlyingType(dataType);
 
-                        int dataTypeSize = System.Runtime.InteropServices.Marshal.SizeOf(dataType);
-                        if (dataTypeSize < sizeof(int))
+                        if (dataType == typeof(byte) || dataType == typeof(sbyte)
```

Comment:
I am wondering whether this should also check for bool and char types that work IL, but not in C#. It is unlikely that these theoretical enum types work with EventSource, so not doing anything about them is probably ok

Reply:
"I chose not to treat `char` and `bool`, but we could (the only other remaining options)." :).

Since now there's two of us wondering that, let's handle them.

```suggestion
                        if (dataType == typeof(byte) || dataType == typeof(sbyte)
                            || dataType == typeof(bool) || dataType == typeof(char)
```

Reply:
Might be a gotcha: it looks like `bool` and `char` have different marshaling sizes: char [managed: 2, marshalling: 1] and bool [managed: 1, marshalling: 4]. Interestingly, `Marshal.SizeOf<Rune>()` returns 4 (while it returns 1 for `char`). :)

Reply:
Good point. This suggests that we should not bother with char or bool here.

Reply:
Yeah, it's questionable what this is going to do for those. Maybe we're better off without the explicit handling for these?

The bool should probably still take int route (see `The manifest defines a bool as a 32bit type (WIN32 BOOL), not 1 bit as CLR Does.` above where we will land after the `goto Again`) so it doesn't matter whether we normalize or not - and char is less than 4 bytes in either case so we would have normalized previously. The code above hardcodes size of the char to the managed definition (2 bytes).